File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceContainerHighestDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerHighestDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd14
            AppTheme.STANDARD -> R.color.cstd14
            AppTheme.BLUE -> R.color.cbd14
            AppTheme.GREEN -> R.color.cgd14
            AppTheme.RED -> R.color.crd14
            AppTheme.NOTHING -> R.color.cnod14
        }
    }
}

File: lib/layer2_core/dark/e_surface_container_highest_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceContainerHighestDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd14;
      case 'standard': return cstd14;
      case 'blue': return cbd14;
      case 'green': return cgd14;
      case 'red': return crd14;
      case 'nothing': return cnod14;
      default: return cstd14;
    }
  }
}
